### PR TITLE
Increased M2Crypto to 0.30.1; Used M2Crypto on Windows

### PR DESCRIPTION
**Pull request by andy-maier**
For details, see the commit message.
Ready for review and merge.

For the review:
- Try it out on Windows
- Review the whole install process we now have on Windows (see install section)

### CODE REVIEW

**Comment by andy-maier** [comment]
One of the issues involved in this is that choco does not create shim files for swig. I have opened issue chocolatey/shimgen#43 for that.

File: pywbem_os_setup.sh [hunk 1]
```diff
@@ -213,12 +213,13 @@ elif [[ "$distro_family" == "osx" ]]; then
...
-  # binary packages and don't invoke Swig during their installation.
-
+  if [[ "$purpose" == "install" ]]; then
+    echo "$myname: Invoking 'pywbem_os_setup.bat install' on platform ${platform}." >&2
+    cmd /c pywbem_os_setup.bat install
+  fi
   if [[ "$purpose" == "develop" ]]; then
-    echo "$myname: Warning: Installation of OS-level packages for development must be performned manually on platform ${platform}." >&2
-    echo ". See the 'Development' chapter of the documentation for instructions." >&2
+    echo "$myname: Warning: Invoke 'pywbem_os_setup.bat develop' on platform ${platform}." >&2
```
Comment (KSchopmeyer) [inline pywbem_os_setup.sh]: Did you mean Invoking???

Comment (andy-maier) [reply pywbem_os_setup.sh]: yes. done.

**Comment by andy-maier** [comment]
Rebased.

**Comment by KSchopmeyer** [comment]
Note: Since in the Readme we state that it is supported on python 2 and 3 we really need to clarify the readme.

File: appveyor.yml [hunk 2]
```diff
@@ -121,29 +94,41 @@ install:
...
+
+  # Verify that Chocolatey is available
+  - where choco
+  - choco --version
+
+  # Circumvention:
+  # The installation in pywbem_os_setup.bat installs swig but without generating
+  # the swig.exe shim file (using GenShim).
+  # See GenShim issue https://github.com/chocolatey/shimgen/issues/43
+  # Therefore, we install swig ihere in order to create the swig.exe shim file.
```
Comment (KSchopmeyer) [inline appveyor.yml]: What is word ihere??

File: docs/intro.rst [hunk 3]
```diff
@@ -108,7 +108,7 @@ Pywbem is supported in these environments:
 Limitations:
 
 * On Windows (native), pywbem is not supported on Python 2.6, because the
-  M2CryptoWin32/64 packages do not support Python 2.6.
+  M2Crypto package does not support Python 2.6.
```
Comment (KSchopmeyer) [inline docs/intro.rst]: Do you mean the M2Crypto windows package???

File: docs/intro.rst [hunk 4]
```diff
@@ -268,6 +263,8 @@ environment (i.e. without using a UNIX-like environment; for that, see
 
       > pip install pywbem
 
+.. _`Chocolatey package manager`: https://chocolatey.org/
```
Comment (KSchopmeyer) [inline docs/intro.rst]: Should we say more than just Chocolatey package manager and a url???

**Comment by KSchopmeyer** [comment]
I reviewed the files changes but want to go back and completely read the doc.  Also I am fighting chocolatey right now on my windows install and will update you when I get through that.

**Comment by KSchopmeyer** [comment]
Which will be slow since windows just decided to do a major update.

File: appveyor.yml (hunk 2 above)
Comment (andy-maier) [reply appveyor.yml]: "here" with an "i" for vi insert mode ;-)
DONE.

File: docs/intro.rst (hunk 3 above)
Comment (andy-maier) [reply docs/intro.rst]: This PR changes from using the outdated M2CryptoWin32/64 packages to using the M2Crypto package, on Windows.

File: docs/intro.rst (hunk 4 above)
Comment (andy-maier) [reply docs/intro.rst]: This is just the definition of a link target; it does not generate anything in the text. It is used a few lines up.

**Comment by andy-maier** [comment]
On updating the README: I don't think we should update the README to duplicate the complete information from the Installation section. I have updated the README to state that the installation method shown is for Linux and that other operating systems are described in the installation section.

**Comment by andy-maier** [comment]
Open at this point for Karl:

* Review the complete installation section of the documentation
* Verify the documented installation approach on Windows

**Comment by KSchopmeyer** [comment]
Testing with windows.   This is fun.

1. Found that the terminal has to be in admin mode since chocolatey accesses data in the C:/program files ... directory and that fails without admin mode.  That, of course means a difference setup since this is not logical like sudo.

2. pywbem_os-setup.bat requires a parameter develop or install and that is not documented.

3. Got through that and got it to build m2crypto and hit:

   Creating library build\temp.win-amd64-2.7\Release\SWIG\_m2crypto.lib and object build\temp.win- amd64-2.7\Release\SWIG\_m2crypto.exp
   python setup.py bdist_wheel
   usage: setup.py [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
   or: setup.py --help [cmd1 cmd2 ...]
   or: setup.py --help-commands
   or: setup.py cmd --help

error: invalid command 'bdist_wheel'
Error: Command returned rc=1

Quit for night at that point

File: docs/intro.rst (hunk 3 above)
Comment (KSchopmeyer) [reply docs/intro.rst]: So should say that M2Crypto package on Windows does not support ...

**Comment by KSchopmeyer** [comment]
New comments:

Made it all the way through a pip install and clone download etc. on windows and tested by running wbemcli.

1.  I propose that we change "and it is assumed that Chocolatey is installed"  in the windows install section to  "Chocolatey must be installed before running the above script."  Just want to make that stand out since it is a PIA to do.

2. The script asks that either "develop" or "install" keywords be on the cmd line.  a)That probably deserves more explanation and b) that should be in the doc.

3. I could not checkout all the links in the doc (ex. to pywbem_os_bat) because in the local builddoc they are broken.

I cannot explain some of the fits and starts in testing but it did work out when I cleaned everything out and started over.

I think with these comments it is ready to commit.  Then, however, we should go through the testing again since this is so many lose parts all sort of flying in formation.

Congratulations on getting this one sorted out

File: docs/intro.rst (hunk 3 above)
Comment (andy-maier) [reply docs/intro.rst]: That's what it says, if you read the whole sentence.

**Comment by andy-maier** [comment]
This last commit is supposed to address all comments that were made.
If anything is not addressed, please make new comments.

**Comment by andy-maier** [comment]
I had to do a small fix with directory switching in pywbem_os_setup.bat

**Pull request merged**